cypress/integration/articleFlows/mangageFlows/lockDiscussion.spec.js

```diff
@@ -0,0 +1,107 @@
+describe('Lock discussion', () => {
+  const getDiscussionLockButton = () =>
+    cy.findByRole('link', {
+      name: /Lock discussion/,
+    });
+
+  const getDiscussionUnlockButton = () =>
+    cy.findByRole('link', {
+      name: /Unlock discussion/,
+    });
+
+  const getDiscussionLockSubmitButton = () => cy.get('.crayons-btn--danger');
```

Comment:
```suggestion
  const getDiscussionLockSubmitButton = () => cy.findByRole('button', { name: 'Lock discussion' });
```

Just a suggestion, but changing this to another `findByRole` would help us test closer to the user experience (since a user finds by the button name rather than classname)

(same goes for the other test file that uses the same)

Reply:
I originally did this intentionally so these specs don't fail for a quick "text change" on the button. I was thinking we don't mind what the button text is as much as that there is a button there that does what we expect, in general.

Regardless, I implemented your suggestion.

Reply:
I commented on the other similar one below, but the general philosophy with e2e tests is to test as close to how a user would use the app as possible. With that in mind, it makes more sense to identify buttons/links/etc by their names, rather than CSS class or ID attribute because those things aren't exposed to users (and equally could change in the future).

It also helps catch accessibility issues on a page, as `findByRole` (unlike `get`) will only succeed if the button is visible to both the UI and assistive technology - e.g. if I add a `hidden` class to the button, the `get` selector will still return the element whereas `findByRole` will fail. Which is pretty neat!

Just mentioning all this for context 🙂